### PR TITLE
Creature/BossAI/Sapphiron: Fixed minor bugs/quirks

### PR DESCRIPTION
**Pull request by Treeston**
- Sapphiron will now properly re-engage if the previous attempt ended while he was in flight.
- Sapphiron will now properly be hovering while in the air phase instead of standing idly on the ground.

### CODE REVIEW

File: src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp [hunk 1]
```diff
@@ -123,7 +123,12 @@ class boss_sapphiron : public CreatureScript
                 _Reset();
 
                 if (_phase == PHASE_FLIGHT)
+                {
                     ClearIceBlock();
+                    me->SetReactState(REACT_AGGRESSIVE);
+                }
+
+                me->SetFullHealth();
```
Comment (chaodhib) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: Is this necessary?  Doesn't _Reset() take care of that?

Comment (Treeston) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: It doesn't reset react state. Try resetting Sapphiron (vanish etc) while he's in flight phase - proximity aggro will be disabled.

Comment (Shauren) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: using SetFullHealth is not correct, regen after evade needs to be fixed (in the core, not scripts)

Comment (ccrs) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: script should be updated to use UpdateVictim()

Comment (Treeston) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: UpdateVictim() is already used. It's just below `events.Update(diff)` (line 238 in latest).

File: src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp [hunk 2]
```diff
@@ -303,6 +306,7 @@ class boss_sapphiron : public CreatureScript
                             case EVENT_LIFTOFF:
                                 Talk(EMOTE_AIR_PHASE);
                                 me->SetDisableGravity(true);
+                                me->SetHover(true);
```
Comment (Shauren) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: I cannot confirm this without sniffs but from experience I can tell you that true hover is very rarely used - so this is likely to be incorrect

Comment (Treeston) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: What's a better way to tell the client to display the hover idle animation?

Comment (tkrokli) [inline src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp]: Browse through [3.3.5/Ulduar] Miniron's Hover Mode (#15177),  [3.3.5/Ulduar] XT002 Entering the ground (#15032) and  [3.3.5] VMap data is sometimes not correct (#15161) for ideas.

**Pull request merged**